spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarReaderFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.core;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.pulsar.client.api.MessageId;
+import org.apache.pulsar.client.api.PulsarClient;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.api.Reader;
+import org.apache.pulsar.client.api.ReaderBuilder;
+import org.apache.pulsar.client.api.Schema;
+
+import org.springframework.lang.Nullable;
+
+/**
+ * Default implementation of {@link PulsarReaderFactory}.
+ *
+ * @param <T> message type
+ * @author Soby Chacko
+ */
+public class DefaultPulsarReaderFactory<T> implements PulsarReaderFactory<T> {
+
+	private final PulsarClient pulsarClient;
+
+	private final Map<String, Object> readerConfig;
+
+	public DefaultPulsarReaderFactory(PulsarClient pulsarClient) {
+		this(pulsarClient, Collections.emptyMap());
+	}
+
+	public DefaultPulsarReaderFactory(PulsarClient pulsarClient, Map<String, Object> readerConfig) {
+		this.pulsarClient = pulsarClient;
+		this.readerConfig = readerConfig;
+	}
+
+	@Override
+	public Reader<T> createReader(@Nullable List<String> topics, @Nullable MessageId messageId, Schema<T> schema,
+			@Nullable Map<String, Object> properties) throws PulsarClientException {
+		ReaderBuilder<T> readerBuilder = this.pulsarClient.newReader(schema);
+		readerBuilder.topics(topics);
+		readerBuilder.startMessageId(messageId);
+		Map<String, Object> config = new HashMap<>(this.readerConfig);
```

Comment:
These (and therefore the default config props `readerConfig`) are not currently being used to build the reader. 

How do the properties passed into create() differ from the default readerConfig passed into `<init>`?

Reply:
That is certainly a mistake - I will fix it tomorrow. Once again, I was just laying the foundations - this thing is not done by any means.

Reply:
Actually, we don't need these properties (map) in the API, as we only need the instance field-level config map.

Reply:
Cool. Yeh, I was thinking that we did not need to map either.